File: DEV/CEC/CEC.py
import os
import subprocess


def crash(error_id, function_no, error_content, type):
    """
    Ensure error_id and function_no are strings
    Prepare the data to write to the temporary files
    Write the name of the placeholder script to the temporary file
    Write the error message to the temporary file
    Write the name of the placeholder function to the temporary file
    Write the name of the placeholder language to the temporary file
    Write the name of the placeholder crash to the temporary file
    Write the type to the temporary file
    Open Crash_Reporter.py in a new shell window
    """
    # Ensure error_id and function_no are strings
    error_id = str(error_id)
    function_no = str(function_no)

    # Prepare the data to write to the temporary files
    script_name = os.path.basename(__file__)
    language = os.path.splitext(__file__)[1][1:]  # Extracting the language part

    # Write the name of the placeholder script to the temporary file
    with open("flag.temp", 'w') as f:
        f.write(script_name)

    # Write the error message to the temporary file
    with open("error.temp", 'w') as f:
        f.write(error_id)

    # Write the name of the placeholder function to the temporary file
    with open("function.temp", 'w') as f:
        f.write(function_no)

    # Write the name of the placeholder language to the temporary file
    with open("language.temp", 'w') as f:
        f.write(language)

    # Write the name of the placeholder crash to the temporary file
    with open("error_data.temp", 'w') as f:
        f.write(error_content)

    with open("type.temp", 'w') as f:
        f.write(type)

    # Open Crash_Reporter.py in a new shell window
    # Note: This command works for Command Prompt.
    # Adjust according to your needs.
    process = subprocess.Popen(r'powershell.exe -Command "& .\Crash_Reporter.py"', shell=True,  stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
    for line in iter(process.stdout.readline, b''):
        decoded_line = line.decode('utf-8').strip()
        print(decoded_line)
    # Wait for the process to finish and get the final output/error
    stdout, _ = process.communicate()
    # Decode the output from bytes to string
    stdout = stdout.decode('utf-8') if stdout else ""
    print(stdout)


crash("PE", "fun5", "ERR", "crash")  # either crash or anything else

File: DEV/CEC/CEC.c
#include <stdio.h>
#include <stdlib.h>

void crash(const char* errorId, const char* functionName, const char* errorContent, const char* type) {
    FILE* flagFile;
    FILE* errorFile;
    FILE* functionFile;
    FILE* languageFile;
    FILE* errorDataFile;
    FILE* typeFile;

    // Attempt to open each file using fopen_s
    if (fopen_s(&flagFile, "flag.temp", "w")!= 0) {
        fprintf(stderr, "Failed to open flag.temp file\n");
        return;
    }

    if (fopen_s(&errorFile, "error.temp", "w")!= 0) {
        fprintf(stderr, "Failed to open error.temp file\n");
        return;
    }

    if (fopen_s(&functionFile, "function.temp", "w")!= 0) {
        fprintf(stderr, "Failed to open function.temp file\n");
        return;
    }

    if (fopen_s(&languageFile, "language.temp", "w")!= 0) {
        fprintf(stderr, "Failed to open language.temp file\n");
        return;
    }

    if (fopen_s(&errorDataFile, "error_data.temp", "w")!= 0) {
        fprintf(stderr, "Failed to open error_data.temp file\n");
        return;
    }

    if (fopen_s(&typeFile, "type.temp", "w")!= 0) {
        fprintf(stderr, "Failed to open type.temp file\n");
        return;
    }

    // Write the data to the files
    if (fprintf(flagFile, "%s", "PlaceholderScriptName.c") < 0) {
        perror("Error writing to flag.temp");
    }
    fclose(flagFile);

    if (fprintf(errorFile, "%s", errorId) < 0) {
        perror("Error writing to error.temp");
    }
    fclose(errorFile);

    if (fprintf(functionFile, "%s", functionName) < 0) {
        perror("Error writing to function.temp");
    }
    fclose(functionFile);

    if (fprintf(languageFile, "%s", type) < 0) {
        perror("Error writing to language.temp");
    }
    fclose(languageFile);

    if (fprintf(errorDataFile, "%s", errorContent) < 0) {
        perror("Error writing to error_data.temp");
    }
    fclose(errorDataFile);

    if (fprintf(typeFile, "%s", type) < 0) {
        perror("Error writing to type.temp");
    }
    fclose(typeFile);

    // Execute the crash reporter
    system("powershell.exe -Command \"&.\\Crash_Reporter.py\"");
}

int main() {
    crash("PE", "fun5", "ERR", "crash");
    return 0;
}
